Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/manager_app/view/plan_list/PlanListActivity.java

```diff
@@ -40,7 +40,7 @@ public void onRemovePlanClick(long itemId) {
                     toastUserNotifier.displayNotifications(
                             R.string.plan_removed_message,
                             getApplicationContext());
-                    planListAdapter.setPlanItems(planTemplateRepository.getAll());
+                    planListAdapter.setPlanItems(planTemplateRepository.getFilteredByName(searchView.getQuery().toString()));
```

Comment:
Thanks, it works ok now. 

But I do not think this line is really needed now. Look searchView.setQuery(searchView.getQuery(), true); is calling the search listener onQueryTextSubmit by setting submit to true. So it is already reloaded. Sorry I haven't notice it earlier but we are performing the same db query twice when it's not really needed and it won't be good for 
 app's performance.  Please remove this line, 39 is enough. The test is really nice, but there is very similar test: whenSearchPlanIsRemovedExpectItToBeRemoved and it would be good to not duplicate it and  only update this test with check that list was refreshed after delete from search. UI test usually takes much time, because of setting db, clicking and clearing after and having same test cases twice is not efficient. If that would be unit test, it would be all ok, because they are fast and do not required setting all app. So I suggest to update the test **whenSearchPlanIsRemovedExpectItToBeRemoved**

```
 @Test
    public void whenSearchPlanIsRemovedExpectItToBeRemoved(){
    final int testedPlanPosition = 5;

        onView(withId(R.id.menu_search)).perform(typeText(expectedName + testedPlanPosition));

        onView(withId(R.id.rv_plan_list))
                .perform(RecyclerViewActions
                        .actionOnItemAtPosition(0,
                                new ViewClicker(R.id.id_remove_plan)));
       //add only this case  
       closeSoftKeyboard();

        onView(withRecyclerView(R.id.rv_plan_list)
                .atPosition(0))
                .check(doesNotExist());
   /// end
        onView(isAssignableFrom(EditText.class)).perform(clearText());
        closeSoftKeyboard();

        onView(withId(R.id.rv_plan_list)).perform(scrollToPosition(testedPlanPosition));
        onView(withRecyclerView(R.id.rv_plan_list)
                .atPosition(testedPlanPosition))
                    .check(matches(hasDescendant(withText(expectedName
                        + (testedPlanPosition + 1)))));
}
```

Reply:
Got it working, didn't notice that we are already re-searching, thanks. Tests refactored and all green for me on wiped android.